blockstore/apps/api/v1/serializers/snapshots.py

```diff
@@ -0,0 +1,49 @@
+"""
+Serializers for Snapshots.
+"""
+
+import binascii
+from django.core.files.storage import default_storage
+from rest_framework import serializers
+from rest_framework.reverse import reverse
+
+
+class FileHyperlinkedIdentityField(serializers.Field):
+
+    def get_attribute(self, obj):
+        # We pass the object instance onto `to_representation`,
+        # not just the field attribute.
+        return obj
+
+    def to_representation(self, obj):
+
+        kwargs = self.context['detail_view_kwargs']
+        kwargs['path'] = obj.path
+
+        return reverse(
+            self.context['detail_view_name'],
+            kwargs=kwargs,
+            request=self.context['request'],
+        )
+
+
+class FileDataField(serializers.FileField):
+
+    def get_attribute(self, obj):
+        # We pass the object instance onto `to_representation`,
+        # not just the field attribute.
+        return obj
+
+    def to_representation(self, obj):
+        data_location = "{}/data/{}".format(self.context['uuid'], binascii.hexlify(obj.hash_digest).decode('utf-8'))
+        if default_storage.exists(data_location):
+            return self.context['request'].build_absolute_uri(default_storage.url(data_location))
+
+
+class FileSerializer(serializers.Serializer):
+
+    path = serializers.CharField()
+    public = serializers.BooleanField()
+    size = serializers.IntegerField(read_only=True)
+    url = FileHyperlinkedIdentityField(read_only=True)
+    data = FileDataField()
```

Comment:
We could try deriving it from the filename? What do you think clients will want to do with the information?

Reply:
If the file is to be made available for download or viewing in a browser, then knowing the mimetype is useful so that it can be served with the appropriate Content-Type and/or Content-Disposition headers.  Various cloud storage folks have APIs to store this info in their object storage protocols (e.g [Google Cloud](https://cloud.google.com/storage/docs/metadata), [OpenStack](https://developer.openstack.org/api-ref/object-store/#create-update-or-delete-container-metadata), [AWS S3](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUT.html)).

The mimetype information can be inferred from the filename (unreliably) or better yet, from the file content itself, and so it makes some sense to store/update this when the file data itself is received.  But I don't know if it's better to store this here in the in Blockstore model, or let it be metadata that's set elsewhere.

Reply:
Hmm. Yeah, it makes sense to have it stored as part of the metadata somewhere. We will need it for cases where for example a web file manager wants to decide which preview component to pass the file URL too. What it be useful to store the modified date or any other metadata that file managers commonly need?

Reply:
Ah yep, last modified date, content type, whether it should be served inline or as an attachment -- these are all potentially useful pieces of metadata.

Worth thinking about, but I see they're not relevant to this ticket.  The needs for this data will arise out of subsequent use cases, when we start trying to serve out these bundle files to browsers.  So long as it's OK to expand the bundle data model to suit these, I'm fine to let this thread go for now.